app/views/report_templates/generate.html.erb

```diff
@@ -11,7 +11,16 @@
                 (_('This will generate a report %s. Based on its definition, it can take a long time to process.') % h(@template.name)) +
                 '</p>').html_safe) %>
 
-  <%= text_f f, :generate_at, label: _('Generate at'), label_help: _('Generate report on a given time. Both date and time are required. Please use format yyyy-mm-dd HH:MM +000') %>
+  <div id="report_generate_at"></div>
+  <%= mount_react_component('DateTime', '#report_generate_at', {
+      label: _('Generate at'),
+      label_help: _('Generate report on a given time.'),
+      id: 'report_generate_at_datetime',
+      value: ( f.object.generate_at || Time.now ).try(:iso8601),
```

Comment:
Is the try really needed? Doesn't `Time.now` guarantee it's not nil?

Reply:
You are right, thanks.

Reply:
Well, it can be empty string, can't it? If I submit the form without specifying it and get back to form due to validation error. Empty string is not falsy value. I don't mind being defensive here.